osler/inventory/views.py

```diff
@@ -110,12 +118,21 @@ def form_valid(self, form):
         df.save()
         return redirect('inventory:drug-list')
 
+
 def drug_dispense(request):
     pk = request.POST['pk']
     num = request.POST['num']
-    drug = get_object_or_404(models.Drug, id=pk)
+    patient_pk = request.POST['patient_pk']
```

Comment:
Is this done through a form? If we embedded a form inside the modal, it would manage a lot of this for us I think.